packages/flutter/lib/src/cupertino/sheet.dart

```diff
@@ -184,21 +192,28 @@ Future<T?> showCupertinoSheet<T>({
                       }
                       Navigator.of(context, rootNavigator: true).pop(result);
                     },
-                    child: effectivePageBuilder!(context),
+                    child: child,
                   );
                 },
               ),
             ];
           },
         ),
       );
-    };
-  }
+    }
 
-  return Navigator.of(
-    context,
-    rootNavigator: true,
-  ).push<T>(CupertinoSheetRoute<T>(builder: widgetBuilder, enableDrag: enableDrag));
+    final PageRoute<T> route = effectiveBuilder != null
```

Comment:
Can you simplify the logic to de-dupe and call on your new wrapper method based on `useNestedNavigation`?